src/Equinox.CosmosStore/CosmosStore.fs

```diff
@@ -565,33 +565,43 @@ module internal Sync =
 
 module Initialization =
 
-    type [<RequireQualifiedAccess>] Provisioning = Container of rus: int | Database of rus: int | Serverless
-    let adjustOfferC (c:Container) (rus : int) = async {
+    // Note: the Cosmos SDK does not (currently) support changing the Throughput mode of an existing Database or Container.
+    type [<RequireQualifiedAccess>] Throughput = Manual of rus: int | Autoscale of maxRus: int
+    type [<RequireQualifiedAccess>] Provisioning = Container of Throughput | Database of Throughput | Serverless
+    let toThroughputProperties = function
+        | Some (Throughput.Manual rus) -> ThroughputProperties.CreateManualThroughput(rus)
+        | Some (Throughput.Autoscale maxRus) -> ThroughputProperties.CreateAutoscaleThroughput(maxRus)
+        | _ -> null
+    let adjustOfferC (c:Container) (throughput : Throughput) = async {
+        let tp = Some throughput |> toThroughputProperties
         let! ct = Async.CancellationToken
-        let! _ = c.ReplaceThroughputAsync(rus, cancellationToken = ct) |> Async.AwaitTaskCorrect in () }
-    let adjustOfferD (d:Database) (rus : int) = async {
+        let! _ = c.ReplaceThroughputAsync(tp, cancellationToken = ct) |> Async.AwaitTaskCorrect in () }
+    let adjustOfferD (d:Database) (throughput : Throughput) = async {
+        let tp = Some throughput |> toThroughputProperties
         let! ct = Async.CancellationToken
-        let! _ = d.ReplaceThroughputAsync(rus, cancellationToken = ct) |> Async.AwaitTaskCorrect in () }
-    let private createDatabaseIfNotExists (client:CosmosClient) dName maybeRus = async {
+        let! _ = d.ReplaceThroughputAsync(tp, cancellationToken = ct) |> Async.AwaitTaskCorrect in () }
+    let private createDatabaseIfNotExists (client:CosmosClient) dName maybeThroughput = async {
+        let tp = maybeThroughput |> toThroughputProperties
```

Comment:
```suggestion
        let tp = maybeThroughput |> Option.map toThroughputProperties |> Option.toObj
```

Reply:
Hope you're ok with my alternate less explicit way of resolving it @deviousasti  ;)